ludwig/features/text_feature.py

```diff
@@ -293,6 +301,24 @@ def update_config_with_metadata(feature_config, feature_metadata, *args, **kwarg
                     "for feature {}".format(feature_config.column)
                 )
 
+    def update_metrics(self, targets: Tensor, predictions: Dict[str, Tensor]) -> None:
+        """Updates metrics with the given targets and predictions.
+
+        Args:
+            targets: Tensor with target values for this output feature.
+            predictions: Dict of tensors returned by predictions().
+        """
+        for metric_name, metric_fn in self._metric_functions.items():
+            # Skip loss metric for LLMs, since we don't compute it in the forward pass
+            # Perplexity needs to be computed from "probabilities" tensor, but right now
+            # we're only producing a "predictions" tensor as output from the forward pass.
+            if self.is_llm_task and metric_name in ("loss", "perplexity"):
```

Comment:
Same as above, would be great if we could avoid all this special casing. For perplexity, can just similarly give it an arbitrary value for now.